### PR TITLE
Reconciler: Some refactoring

### PR DESCRIPTION
**Pull request by tmtmtmtm**
The logic for showing the cards on the left and right side (incoming vs
existing people) was getting a little hard to follow, as they were both
using the same template, even though they were conceptually quite
different things, and almost every section of the card had to be wrapped
in a check to see which type it was.

Split these out so that it's easier to follow (even though there’s now a
small amount of duplication between them), but also harmonise some of
the other code that was needlessly different.

### CODE REVIEW

**Comment by chrismytton** [comment]
Yep, this is much easier to follow with these two template split up. I think it was probably the case that they were very similar to start with and then drifted apart. :+1: :ship:

**Comment by tmtmtmtm** [comment]
Yeah — that's a pattern I've noticed a few times now. We start out doing a very simple version of something, and end up with two things that look close to identical, so abstract them out into a common form. But then as we add more and more code later it becomes increasingly obvious that the things were actually quite different — but there's never a key point where it's obvious enough that we should split them up rather than adding just a single additional `if`…
